internal/dag/gatewayapi_processor.go

```diff
@@ -332,6 +333,10 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRo
 				continue
 			}
 
+			// Apply the weight of the forwardTo.
+			service.Weighted.Weight = uint32(forward.Weight)
```

Comment:
From gateway api spec:

"If unspecified, weight defaults to 1."

Do a nil check, and if so, set to 1?

Reply:
We're still based off of v0.2.0 where weight isn't nil, but has since changed in `master` branch. I don't understand why we have this as a nil with a default. To me, if it's not defined, you get `1` as the default so is it really going to be nil? 

We'll need to update this when we move to work off of master which should come next (https://github.com/projectcontour/contour/issues/3578).